### ISSUE TITLE
BUG - Parsing of markdown is not correct for code section (easily reproducible)

### ISSUE THREAD

**Issue by lfmunoz** [markdown-it/markdown-it#960]
The following markdown does not parse correctly:

```

# Some Markdown Documentation 

This markdown does not parse correctly 

<details>
<summary> <strong> cat src/predefined_vars.sh </strong> </summary>
<p class="block">
\```bash         NOTE:  I added \ so you can view this here in GITHUB
#!/bin/bash

# This describes the shell that will be interpreting any commands you type in. 
#  In most cases, this will be bash by default, but other values can be set if 
#  you prefer other options.
echo $SHELL
# /usr/bin/zsh
\```            NOTE:  I added \ so you can view this here in GITHUB
</p></details>
```

It calls  md.options.highlight = (str, lang) => { }

with 
```
lang = 
str = "</p></details>"
```

if I put a space after the line `<p class="block">` it works

Why does the parser need a new line between html the the start of markdown code block?

Here is the problem reproduced

https://markdown-it.github.io/#md3=%7B%22source%22%3A%22%23%20Some%20Markdown%20Documentation%20%5Cn%5CnThis%20markdown%20does%20not%20parse%20correctly%20%5Cn%5Cn%3Cdetails%3E%5Cn%3Csummary%3E%20%3Cstrong%3E%20cat%20src%2Fpredefined_vars.sh%20%3C%2Fstrong%3E%20%3C%2Fsummary%3E%5Cn%3Cp%20class%3D%5C%22block%5C%22%3E%5Cn%5Cn%60%60%60bash%20%20%20%20%5Cn%23!%2Fbin%2Fbash%5Cn%5Cn%23%20This%20describes%20the%20shell%20that%20will%20be%20interpreting%20any%20commands%20you%20type%20in.%20%5Cn%23%20%20In%20most%20cases%2C%20this%20will%20be%20bash%20by%20default%2C%20but%20other%20values%20can%20be%20set%20if%20%5Cn%23%20%20you%20prefer%20other%20options.%5Cnecho%20%24SHELL%5Cn%23%20%2Fusr%2Fbin%2Fzsh%5Cn%60%60%60%20%20%20%20%20%20%5Cn%3C%2Fp%3E%3C%2Fdetails%3E%22%2C%22defaults%22%3A%7B%22html%22%3Atrue%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22html%22%7D%7D

remove new line after `<p class="block">`  to see issue

**Comment by puzrin** [comment]
> Why does the parser need a new line between html the the start of markdown code block?

Because this behavior described in CM spec. Please check everything about markdown in reference parser first https://spec.commonmark.org/dingus/, and see spec examples.

**Comment by lfmunoz** [comment]
@puzrin Thanks for looking into this.  Doesn't it seem like  ridiculous behavior to you?  Is newline verses no new line really that hard to parse? If the spec is doing something that doesn't make sense doesn't mean you have to follow it exactly?

**Comment by puzrin** [comment]
Spec has a forum to suggest improvements, I can't discuss things about markup. Also, you can write your custom plugin and replace the default becaviour.
